docs/variants/dell_optiplex/initial-deployment.md

```diff
@@ -76,7 +76,97 @@ page](../../dasharo-tools-suite/releases.md). Once you have obtained it, you can
 then proceed with following the [Dasharo zero-touch initial deployment
 section](../../dasharo-tools-suite/documentation.md#dasharo-zero-touch-initial-deployment)
 procedure. This will help you set up Dasharo effectively and without manual
-intervention.
+intervention. However, if you wish to flash firmware you have built yourself,
+you will need take an alternate route and go through firmware preparation.
+
+## Firmware preparation
+
+A fresh firmware binary built from source for this platform is not necessarily
+complete and ready to be deployed on hardware.
+
+If you have built it using the non-TXT config, you can safely go ahead and
+flash, the platform will be fully operational - but due to lack of EC firmware,
+the fans will always spin at full speed, which can be a nuisance.
+
+However, if you have build the firmware with TXT support, without patching the
+.rom with proprietary blobs the platform will simply refuse to boot.
+
+This brings us to patching the binary. This guide covers the process of
+obtaining 3 firmware blobs necessary for a full-feature deployment on the
+OptiPlex platform:
+
+* `sch5545_ecfw.bin` - EC firmware
+* `IVB_BIOSAC_PRODUCTION.bin` - Ivy Bridge BIOS ACM - **TXT support only**
+* `SNB_IVB_SINIT_20190708_PW.bin` - Ivy Bridge SINIT ACM - **TXT support only**
+
+It is recommended you perform this process booted into the Dasharo Tools Suite,
+unless you are willing to install all of the reqiuired utilities yourself.
+
+### Obtaining the blobs
+
+The process consists of downloading the stock BIOS via an update link and
+extracting all of its contents, to then salvage what we need for our firmware.
+
+First, you will need to download the stock BIOS image from an update URL. As
+of the moment of writing this guide, the working link for OptiPlex 7010/9010
+is [this](https://dl.dell.com/FOLDER05066036M/1/O7010A29.exe).
+
+Then, extract all of the components using `binwalk`:
+
+```bash
+sudo binwalk -e "BIOS_UPDATE_FILENAME -C ."
+```
+
+Assuming the extraction process was successful, you should now have an
+extracted UEFI image file, hidden under an unassuming name such as
+`65C10`. Deeper down the rabbit hole, you will now have to extract the
+blobs from this image using `uefi-firmware-parser`:
+
+```bash
+ uefi-firmware-parser -e "UEFI_IMAGE_FILE" -O
+```
+
+Congratulations, you should now have access to the BIOSAC and EC firmware
+files. Now, only to find them. The expected paths to each file are as follows:
+
+* EC firmware -
+  `_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-
+  a316-22dd0517c1e8/volume-0x50000/file-d386beb8-4b54-4e69-94f5-06091f67e0d3/
+  section0.raw`
+* BIOS ACM file -
+  `_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-
+  a316-22dd0517c1e8/volume-0x500000/file-2d27c618-7dcd-41f5-bb10-21166be7e143/
+  object-0.raw`
```

Comment:
It is not clear how one reaches this result, because you used the variables. Would you be able to reproduce it based on this initial deployment guide? I would have problems myself.

Reply:
Fixed in [this commit](https://github.com/Dasharo/docs/pull/873/commits/2d83ab4937b0494d95d8012a011eb1fc4e69fefa)